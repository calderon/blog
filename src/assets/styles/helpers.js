import { css } from "styled-components"

const srOnly = css`
  position: absolute !important;
  width: 1px !important;
  height: 1px !important;
  padding: 0 !important;
  margin: -1px !important;
  overflow: hidden !important;
  clip: rect(0, 0, 0, 0) !important;
  white-space: nowrap !important;
  border: 0 !important;
`

const srOnlyFocusable = css`
  &:not(:focus) {
    ${srOnly};
  }
`

const skipLink = css`
  ${srOnly};
  ${srOnlyFocusable};
`

const _breakpoints = {
  xs: 480,
  sm: 640,
  md: 768,
  lg: 840,
  xl: 1024,
}

const devices = {
  xs: `(min-width: ${_breakpoints.xs}px)`,
  sm: `(min-width: ${_breakpoints.sm}px)`,
  md: `(min-width: ${_breakpoints.md}px)`,
  lg: `(min-width: ${_breakpoints.lg}px)`,
  xl: `(min-width: ${_breakpoints.xl}px)`,
}

export { srOnly, srOnlyFocusable, skipLink, devices }
