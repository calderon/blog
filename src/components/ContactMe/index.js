import React from "react"

import {
  Container,
  ContainerHeader,
  ContainerHeading,
  ContainerMain,
  ContainerColumn,
} from "../Container"

const ContactMe = props => {
  return (
    <Container as="aside" id="contact" main primary>
      <ContainerHeader>
        <ContainerHeading>Contact me</ContainerHeading>
      </ContainerHeader>
      <ContainerMain columnize={3}>
        <ContainerColumn>
          <p>Take a sit. How can I help you?</p>
        </ContainerColumn>
        <ContainerColumn>Your name + Email</ContainerColumn>
        <ContainerColumn>Your message + Send your message</ContainerColumn>
      </ContainerMain>
    </Container>
  )
}

export default ContactMe
