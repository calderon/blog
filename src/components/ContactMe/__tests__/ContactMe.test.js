import React from "react"
import { render } from "../../../../__tests__/helpers/render"

import ContactMe from "../"

const renderComponent = ({ ...props } = {}) => render(<ContactMe {...props} />)

describe("ContactMe", () => {
  it("renders correctly", () => {
    const contactMe = renderComponent()
    expect(contactMe).toMatchSnapshot()
  })
})
