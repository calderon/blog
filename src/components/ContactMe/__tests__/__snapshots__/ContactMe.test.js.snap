// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMe renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <aside
        class="sc-AxirZ bfdHoz"
        id="contact"
      >
        <header
          class="sc-AxiKw ihVPJd"
        >
          <h1
            class="sc-AxjAm sc-AxhCb hqTGCv heading heading--h1"
          >
            Contact me
          </h1>
        </header>
        <div
          class="sc-AxhUy jSnPkM"
        >
          <div
            class="sc-AxgMl JpLPp"
          >
            <p>
              Take a sit. How can I help you?
            </p>
          </div>
          <div
            class="sc-AxgMl JpLPp"
          >
            Your name + Email
          </div>
          <div
            class="sc-AxgMl JpLPp"
          >
            Your message + Send your message
          </div>
        </div>
      </aside>
    </div>
  </body>,
  "container": <div>
    <aside
      class="sc-AxirZ bfdHoz"
      id="contact"
    >
      <header
        class="sc-AxiKw ihVPJd"
      >
        <h1
          class="sc-AxjAm sc-AxhCb hqTGCv heading heading--h1"
        >
          Contact me
        </h1>
      </header>
      <div
        class="sc-AxhUy jSnPkM"
      >
        <div
          class="sc-AxgMl JpLPp"
        >
          <p>
            Take a sit. How can I help you?
          </p>
        </div>
        <div
          class="sc-AxgMl JpLPp"
        >
          Your name + Email
        </div>
        <div
          class="sc-AxgMl JpLPp"
        >
          Your message + Send your message
        </div>
      </div>
    </aside>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
